test/es6ToJSONTest.js

```diff
@@ -0,0 +1,220 @@
+      const json = getJSON('CodeableConceptObjectValueEntry');
+      const entry = CodeableConceptValueEntry.fromJSON(json);
+      expect(entry).instanceOf(CodeableConceptValueEntry);
+      
+      let gen_json = entry.toJSON();
+      validateJSON('CodeableConceptObjectValueEntry', gen_json);
+    });
+  });
+
+  describe('#ElementValueEntryClass()', () => {
+    const ElementValueEntry = importResult('shr/simple/ElementValueEntry');
+    it('should serialize a JSON instance', () => {
+      const json = getJSON('ElementValueEntry');
+      const entry = ElementValueEntry.fromJSON(json);
+      expect(entry).instanceOf(ElementValueEntry);
+      
+      let gen_json = entry.toJSON();
+      validateJSON('ElementValueEntry', gen_json);
+    });
+  });
+
+  describe('#RecursiveEntryClass()', () => {
+    const RecursiveEntry = importResult('shr/simple/RecursiveEntry');
+    it('should deserialize a JSON instance', () => {
+      const json = getJSON('RecursiveEntry');
+      const entry = RecursiveEntry.fromJSON(json);
+      expect(entry).instanceOf(RecursiveEntry);
+      let gen_json = entry.toJSON();
+      validateJSON('RecursiveEntry', gen_json);
+      
+      // Recursive child 1
+      const child1 = entry.recursiveEntry[0];
+      expect(child1).instanceOf(RecursiveEntry);
+      let child1_json = child1.toJSON();
+      validateJSON('RecursiveEntry', child1_json);
+
+      // Recursive grandchild 1
+      const grandchild1 = child1.recursiveEntry[0];
+      expect(grandchild1).instanceOf(RecursiveEntry);
+      let grandchild1_json = grandchild1.toJSON();
+      validateJSON('RecursiveEntry', grandchild1_json);
+
+      // Recursive child 2
+      const child2 = entry.recursiveEntry[1];
+      expect(child2).instanceOf(RecursiveEntry);
+      let child2_json = child2.toJSON();
+      validateJSON('RecursiveEntry', child2_json);
+    });
+  });
+
+  describe('#ReferenceEntryClass()', () => {
+    const ReferenceEntry = importResult('shr/simple/ReferenceEntry');
+    it('should serialize a JSON instance', () => {
+      const json = getJSON('ReferenceEntry');
+      const entry = ReferenceEntry.fromJSON(json);
+      expect(entry).instanceOf(ReferenceEntry);
+      
+      let gen_json = entry.toJSON();
+      validateJSON('ReferenceEntry', gen_json);
+    });
+  });
+
+  describe('#BasedOnIntegerValueElementEntryClass()', () => {
+    const BasedOnIntegerValueElementEntry = importResult('shr/simple/BasedOnIntegerValueElementEntry');
+    it('should serialize a JSON instance', () => {
+      const json = getJSON('BasedOnIntegerValueElementEntry');
+      const entry = BasedOnIntegerValueElementEntry.fromJSON(json);
+      expect(entry).instanceOf(BasedOnIntegerValueElementEntry);
+      
+      let gen_json = entry.toJSON();
+      validateJSON('BasedOnIntegerValueElementEntry', gen_json);
+    });
+
+    describe('#OverrideBasedOnIntegerValueElementEntryClass()', () => {
```

Comment:
Is this intentionally nested in the `#BasedOnIntegerValueElementEntryClass` test case?

Reply:
Nope. Thanks for catching this!